adapters/appnexus/appnexus.go

```diff
@@ -462,7 +475,18 @@ func getMediaTypeForBid(bid *openrtb.Bid) (openrtb_ext.BidType, error) {
 	case 3:
 		return openrtb_ext.BidTypeNative, nil
 	default:
-		return "", fmt.Errorf("Unrecognized bid_ad_type in response from appnexus: %d", impExt.Appnexus.BidType)
+		return "", fmt.Errorf("Unrecognized bid_ad_type in response from appnexus: %d", bid.Appnexus.BidType)
+	}
+}
+
+// getIabCategoryForBid maps an appnexus brand id to an IAB category.
+func (a *AppNexusAdapter) getIabCategoryForBid(bid *appnexusBidExt) (string, error) {
+	// TODO: Change from BrandId to a CategoryId once that is returned from impbus
+	brandIDString := strconv.Itoa(bid.Appnexus.BrandId)
+	if iabCategory, ok := a.iabCategoryMap[brandIDString]; ok {
```

Comment:
we should check if read from a nil map is successful.

Reply:
Ok I checked to make sure on the golang.org site it says this: `A nil map behaves like an empty map when reading, but attempts to write to a nil map will cause a runtime panic; don't do that.` So since we're doing a read here if iabCategoryMap is nil it should result in iabCategory="" and ok=false.